src/components/core/geoChart/geoChartDataSource.ts

```diff
@@ -139,3 +144,47 @@ export const createPushpinDataSource = (
     }
     return source;
 };
+
+export const calculateLegendData = (
+    executionResult: Execution.IExecutionResult,
+    geoData: IGeoData,
+): IGeoChartLegendData => {
+    const { size, color } = geoData;
+    const { data } = executionResult;
+    if (isTwoDimensionsData(data)) {
+        const [colorSeries, sizeSeries] = [color, size]
+            .map((item: IGeoDataItem): Execution.DataValue[] => (item ? data[item.index] : []))
+            .map((series: Execution.DataValue[]) => without(series.map(stringToFloat), null, undefined, NaN));
```

Comment:
Could we use https://lodash.com/docs/4.17.15#compact?

Reply:
series may contain `0`